File: src/services/auth_service.ts
import { userApi } from '../api/user_api';
import { tokenService } from './token_service';
import { AuthResponse } from '../types/types';

export const authService = {
  login: async (email: string, password: string): Promise<AuthResponse> => {
    const response = await userApi.login(email, password);
    if (response) {
      tokenService.saveAccessToken(response.accessToken);
      tokenService.saveRefreshToken(response.refreshToken);
      return response;
    }
    throw new Error('Login failed');
  },
  
  logout: () => {
    tokenService.clearTokens();
  },

  checkAuth: async () => {
    const token = tokenService.getAccessToken();
    if (!token) return null;
    
    try {
      // In a real app, you would verify the token with the backend
      // and maybe get the updated user profile.
      // For now, we'll just check if it exists.
      return await userApi.getCurrentUser();
    } catch (error) {
      tokenService.clearTokens();
      return null;
    }
  }
};

File: lib/services/auth_service.dart
import '../api/user_api.dart';
import 'token_service.dart';
import '../types/types.dart';

class AuthService {
  static Future<AuthResponse> login(String email, String password) async {
    final response = await UserApi.login(email, password);
    if (response != null) {
      await TokenService.saveAccessToken(response.accessToken);
      await TokenService.saveRefreshToken(response.refreshToken);
      return response;
    }
    throw Exception('Login failed');
  }

  static Future<AuthResponse> register(String email, String password) async {
    final response = await UserApi.register(email, password);
    if (response != null) {
      await TokenService.saveAccessToken(response.accessToken);
      await TokenService.saveRefreshToken(response.refreshToken);
      return response;
    }
    throw Exception('Registration failed');
  }

  static Future<void> logout() async {
    await TokenService.clearTokens();
  }

  static Future<User?> checkAuth() async {
    final token = await TokenService.getAccessToken();
    if (token == null) return null;
    
    try {
      return await UserApi.getCurrentUser();
    } catch (e) {
      await TokenService.clearTokens();
      return null;
    }
  }

  static Future<bool> forgotPassword(String email) async {
    return await UserApi.forgotPassword(email);
  }

  static Future<bool> resetPassword(String token, String password) async {
    return await UserApi.resetPassword(token, password);
  }
}
